heron/common/src/cpp/network/baseserver.cpp

```diff
@@ -106,6 +106,17 @@ sp_int32 BaseServer::Start_Base() {
     return -1;
   }
 
+  // fetch the port after bind/listen port 0
+  if (AF_INET == options_.get_sin_family() && 0 == options_.get_port()) {
+    if (getsockname(listen_fd_, serv_addr, &sockaddr_len) != 0)  {
```

Comment:
It is better not to put this piece of code in a separate function:
1. This code is not re-used in the other places.
2. Making it separate breaks the same style in the Start_Base()
3. Many variables involve in this code. It is not good tradeoff to pass many variables to a simple separate function.
4. If this code was in a separate function, why would not the other parts of the same Start_Base() be refactored into separate functions?